ToDoListApp/ViewController.swift

```diff
@@ -19,18 +19,51 @@ class ViewController: UIViewController {
         self.title = "Tasks"
         tableView.delegate = self
         tableView.dataSource = self
+        
+        //set the user defaults
+        if !UserDefaults().bool(forKey: "setup"){
+            UserDefaults().set(true, forKey: "setup")
+            UserDefaults().set(0, forKey: "count")
+        }
+        
+        updateTasks()
+    }
+    
+    func updateTasks(){
+        tasks.removeAll()
+        guard let count = UserDefaults().value(forKey: "count") as? Int else{
+            return
+        }
+        
+        for x in 0..<count{
+            if let task = UserDefaults().value(forKey: "task_\(x+1)") as? String {
+                tasks.append(task)
+            }
+        }
+        
+        tableView.reloadData()
     }
 
     @IBAction func didTabAdd(){
         let vc = storyboard?.instantiateViewController(withIdentifier: "entry") as! EntryViewController
         vc.title = "New Task"
+        vc.update = {
+            DispatchQueue.main.async {
```

Comment:
Try without `DispatchQueue.main.async ` and understand the behaviour

Reply:
Try without DispatchQueue show error
UI API called on a background thread

viewDidLoad is already running on the main queue 
DispatchQueue.main.async is when you have code running on a background queue and you need a specific block of code to be executed on the main queue